### PR TITLE
Improve password expiry error message for API based Authentication

### PR DESCRIPTION
**Pull request by HasiniSama**
### Proposed changes in this pull request

- **Fixes**: https://github.com/wso2/product-is/issues/20966

**Previous message:**

```
{
    "flowId": "4ac71e18-52ba-4b11-86d5-f64da810a273",
    "flowStatus": "INCOMPLETE",
    "flowType": "AUTHENTICATION",
    "nextStep": {
        "stepType": "AUTHENTICATOR_PROMPT"
    },
    "links": [
        {
            "name": "authentication",
            "href": "https://localhost:9443/oauth2/authn",
            "method": "POST"
        }
    ]
}
```

**New message:**
Screenshot 2024-09-06 at 17 18 05

### When should this PR be merged

- https://github.com/wso2/carbon-identity-framework/pull/5925

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java [hunk 1]
```diff
@@ -43,13 +43,17 @@
...
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 import java.util.List;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 public class EnforcePasswordResetAuthenticationHandler extends AbstractPostAuthnHandler {
 
     private static final Log log = LogFactory.getLog(EnforcePasswordResetAuthenticationHandler.class);
+    private static final String PASSWORD_EXPIRED_MSG = URLEncoder.encode(
```
Comment (ThaminduR) [inline components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java:55]: Any particular reason to define this constant? Can't we have this inline?

Comment (HasiniSama) [reply components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java:55]: One reason is to future-proof the password expired message by making it easier to modify or reuse.

Comment (HasiniSama) [reply components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java:55]: Also, we avoid repeated calls to the encode function.

**Review by pamodaaw — APPROVED** [review]

**Review by ThaminduR — APPROVED** [review]

File: components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java [hunk 2]
```diff
@@ -43,13 +43,17 @@
...
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 import java.util.List;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 public class EnforcePasswordResetAuthenticationHandler extends AbstractPostAuthnHandler {
 
     private static final Log log = LogFactory.getLog(EnforcePasswordResetAuthenticationHandler.class);
+    private static final String ENCODED_PASSWORD_EXPIRED_MSG = URLEncoder.encode(
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java:55]: Shall we check whether we can move this to constant file?

Comment (HasiniSama) [reply components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/EnforcePasswordResetAuthenticationHandler.java:55]: Updated via: https://github.com/wso2-extensions/identity-governance/pull/950

**Pull request merged**